### PR TITLE
feat: allow sending denoms with URL encoding

### PR DESCRIPTION
**Pull request by emidev98**
Enable querying token factory with [URL encoded](https://www.w3schools.com/tags/ref_urlencode.ASP) token denom, that way LCD can be queried without failing due the following error:
```
{
  "code": 12,
  "message": "Not Implemented",
  "details": []
}
```
This will not cause issues because denoms aren't allowed to have `%` which will no create any collisions

### LCD Query;
image

### CLI Query:
image

based on: https://github.com/terra-money/core/pull/207

### CODE REVIEW

**Comment by mattverse** [comment]
@emidev98 Thanks for the PR! Can you fix the lint please? 🙏

**Comment by emidev98** [comment]
Hello @mattverse, for some reason the lint command does not work in my local. Could you run the linter please? Codewoners can push to the branch. Thanks!

File: x/tokenfactory/keeper/grpc_query.go [hunk 1]
```diff
@@ -19,7 +19,10 @@ func (k Keeper) Params(ctx context.Context, req *types.QueryParamsRequest) (*typ
 
 func (k Keeper) DenomAuthorityMetadata(ctx context.Context, req *types.QueryDenomAuthorityMetadataRequest) (*types.QueryDenomAuthorityMetadataResponse, error) {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
-
+	decodedDenom, err := url.QueryUnescape(req.Denom)
+	if err == nil {
+		req.Denom = decodedDenom
```
Comment (mattverse) [inline x/tokenfactory/keeper/grpc_query.go:24]: Can we add test case for this fix please? 👍

Comment (mattverse) [reply x/tokenfactory/keeper/grpc_query.go:24]: @emidev98 Hey, apologies, totally missed your last comment

**Comment by mattverse** [comment]
@emidev98 Happy to merge this once lint is fixed!

**Review by ValarDragon — APPROVED** [review]

**Comment by ValarDragon** [comment]
Err we need to ensure we didn't introduce any serialization ambiguity here. We should be safe as long as:
- URL escape only works with replacing input with a `%`
- Chain denoms can't have `%` characters.

I feel uncomfortable relying on that first assumption in such a load bearing way though. Could we instead just do a find replace of `%2F` with `/`

**Review by mattverse — APPROVED** [review]
Ah even better :) LGTM

**Comment by emidev98** [comment]
Thanks @mattverse I have been very busy this days and didn't paid attention to some PRs.
I hope this helps you! 🥂

**Comment by mattverse** [comment]
@ValarDragon Feel free to give secondary review for the change, planning to merge this by EOW!

**Comment by mattverse** [comment]
@emidev98 Definitely does save alot of people's lives :) Thanks so much for the PR :)

**Pull request merged**